validator_worker/src/sentry_interface.rs

```diff
@@ -137,30 +154,39 @@ impl<T: Adapter + 'static> SentryApi<T> {
     pub async fn get_event_aggregates(
         &self,
         after: DateTime<Utc>,
-    ) -> Result<EventAggregateResponse, reqwest::Error> {
-        let whoami = self.adapter.whoami();
+    ) -> Result<EventAggregateResponse, Box<ValidatorWorker>> {
+        let mut adapter = self
+            .adapter
+            .write()
+            .expect("get_event_aggregates: Failed to acquire adapter");
+
+        let whoami = adapter.whoami();
+
         let validator = self
             .channel
             .spec
             .validators
             .into_iter()
             .find(|&v| v.id == whoami);
-        let auth_token = self
-            .adapter
-            .get_auth(validator.unwrap())
-            .expect("Failed to get user auth token");
+
+        let auth_token = adapter.get_auth(validator.unwrap());
+        if let Err(e) = auth_token {
```

Comment:
You could map this error and use `?` and get rid off the `unwrap()` bellow.

Same way you did here: https://github.com/AdExNetwork/adex-validator-stack-rust/pull/143/files/e4739e44f81d4c040f627043aa33bbb160c21a71..a781eb41a68ffa5cd31c3b42bfb2de638f773d29#diff-ae6ae3e6a4852afdf6f5eda167faccd7R189